cmd/gardener-extension-os-coreos/app/app.go

```diff
@@ -28,87 +24,72 @@ import (
 const Name = "os-coreos"
 
 // NewControllerCommand creates a new CoreOS controller command.
-func NewControllerCommand(ctx context.Context) *cobra.Command {
-	var (
-		generalOpts = &controllercmd.GeneralOptions{}
-		restOpts    = &controllercmd.RESTOptions{}
-		mgrOpts     = &controllercmd.ManagerOptions{
-			LeaderElection:          true,
-			LeaderElectionID:        controllercmd.LeaderElectionNameID(Name),
-			LeaderElectionNamespace: os.Getenv("LEADER_ELECTION_NAMESPACE"),
-		}
-		ctrlOpts = &controllercmd.ControllerOptions{
-			MaxConcurrentReconciles: 5,
-		}
-		reconcileOpts = &controllercmd.ReconcilerOptions{}
-
-		heartbeatCtrlOpts = &heartbeatcmd.Options{
-			ExtensionName:        Name,
-			RenewIntervalSeconds: 30,
-			Namespace:            os.Getenv("LEADER_ELECTION_NAMESPACE"),
-		}
-
-		controllerSwitches = controllercmd.NewSwitchOptions(
-			controllercmd.Switch(osccontroller.ControllerName, operatingsystemconfig.AddToManager),
-			controllercmd.Switch(heartbeat.ControllerName, heartbeat.AddToManager),
-		)
-
-		aggOption = controllercmd.NewOptionAggregator(
-			generalOpts,
-			restOpts,
-			mgrOpts,
-			ctrlOpts,
-			controllercmd.PrefixOption("heartbeat-", heartbeatCtrlOpts),
-			reconcileOpts,
-			controllerSwitches,
-		)
-	)
+func NewControllerCommand() *cobra.Command {
+	options := NewOptions()
 
 	cmd := &cobra.Command{
 		Use: "os-coreos-controller-manager",
 
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if err := aggOption.Complete(); err != nil {
-				return fmt.Errorf("error completing options: %w", err)
+			if err := options.optionAggregator.Complete(); err != nil {
+				return fmt.Errorf("error completing options: %s", err)
 			}
 
-			if err := heartbeatCtrlOpts.Validate(); err != nil {
+			if err := options.Validate(); err != nil {
 				return err
 			}
 
-			// TODO: Make these flags configurable via command line parameters or component config file.
-			util.ApplyClientConnectionConfigurationToRESTConfig(&componentbaseconfig.ClientConnectionConfiguration{
-				QPS:   100.0,
-				Burst: 130,
-			}, restOpts.Completed().Config)
+			cmd.SilenceUsage = true
+			return options.run(cmd.Context())
```

Comment:
nit: having the `run` method on the `option` struct feels weird, maybe move into function `run` and pass `options` as argument